library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -69,10 +83,13 @@ class ChuckerInterceptor @JvmOverloads constructor(
             throw e
         }
 
-        val processedResponse = processResponse(response, transaction)
-        collector.onResponseReceived(transaction)
-
-        return processedResponse
+        processResponseMetadata(response, transaction)
+        return multiCastResponseBody(response) { cachedResponseFile ->
+            val buffer = cachedResponseFile?.let { readResponseBuffer(it, response.isGzipped) }
+            cachedResponseFile?.delete()
+            if (buffer != null) processResponseBody(response, buffer, transaction)
```

Comment:
This whole block starting with `return` is too hard to read. Let's rewrite it in more simple way without putting multiple functions in `return`.